src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs

```diff
@@ -27,11 +28,21 @@ internal class HighlightingService : IHighlightingService
         public IEnumerable<TextSpan> GetHighlights(
              SyntaxNode root, int position, CancellationToken cancellationToken)
         {
-            return _highlighters.Where(h => h.Metadata.Language == root.Language)
-                               .Select(h => h.Value.GetHighlights(root, position, cancellationToken))
-                               .WhereNotNull()
-                               .Flatten()
-                               .Distinct();
+            try
+            {
+                return _highlighters.Where(h => h.Metadata.Language == root.Language)
+                                   .Select(h => h.Value.GetHighlights(root, position, cancellationToken))
+                                   .WhereNotNull()
+                                   .Flatten()
+                                   .Distinct();
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+            {
+                // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/763988
+                // We still couldn't identify the root cause for the linked bug above. 
```

Comment:
So why wouldn't we make a SupportSyntaxTree change like we did for the syntactic classifier earlier?

Reply:
We can do that, but I 'd like to see if the fix I made in xaml project actually address the issue for xaml.

Reply:
Ah, so the intent is you want to see if this can be hit, and if so, we'd remove this because we've figured out the root cause?

Reply:
Yup. Or if not, we can remove this after we have done what heejae proposed above.

Reply:
If that's the intent, the comment doesn't say that. It says we couldn't identify a root cause. :smile:

Reply:
I guess I don't see why we wouldn't apply the same SupportSyntaxTree check to the highlighting stuff as well, for consistency...

Reply:
Well, you are probably right. If we are confident that enforcing content-type and document language in workspace would eliminate the root cause of the crash and will remove the NFW once it's done, then it doesn't really matter if we know the particular code path we fixed in XAML can be hit or not.

Reply:
Just to be clear all I'd say is we do two things:

1. Do the same SupportsSyntax check in the caller of this code like we did in the other place; until we do the larger API change that @heejaechang was thinking of, let's guard it in both places.
2. Change the comment in this NFW to say that we think the other fix does fix it, and if we don't see any further hits after a certain date we can just delete the NFW. Or leave the NFW in because we're saying as a general policy this feature isn't worth crashing the product over. But in that case, let's revise the comment to make it independent of the bug.

My biggest worry here is as the PR stands somebody looking at this code a few years from now won't be clear why the NFW is there, or why they can't delete if it the bug was indeed fixed.

Reply:
+1 - Agree with @jasonmalinowski.

Reply:
If we added the SupportSyntax check in the caller, then the NFW wouldn't be needed at all.  I will do (1) you asked and add some comment there, then remove NFW